common/lib/xmodule/xmodule/modulestore/xml.py

```diff
@@ -381,17 +381,18 @@ def try_load_course(self, course_dir, course_ids=None, target_course_id=None):
             errorlog.tracker(msg)
             self.errored_courses[course_dir] = errorlog
             monitor_import_failure(target_course_id, 'Updating', exception=exc)
-
-        if course_descriptor is None:
-            pass
-        elif isinstance(course_descriptor, ErrorBlock):
-            # Didn't load course.  Instead, save the errors elsewhere.
-            self.errored_courses[course_dir] = errorlog
-        else:
-            self.courses[course_dir] = course_descriptor
-            course_descriptor.parent = None
-            course_id = self.id_from_descriptor(course_descriptor)
-            self._course_errors[course_id] = errorlog
+            raise exc
+        finally:
```

Comment:
why you are adding this code in `finally`? Do you want this code to be executed in case of an exception? 
What about `try/catch/else`?

Reply:
Yes as per the previous code this was executing even in case of an exception because it was not raised again. So now I have raised the exception after catching it so i have added `finally` block to execute code even in case of exception

Reply:
finally is called regardless of try block result. It will be called for both success and error scenarios. on the other hand `else` is a little different. 

So as you mentioned it was being only executed in case of exception i think you need to reconsider using `finally`

Reply:
As I have mentioned above that block executes even in case of an exception, so `finally` is the right choice because we want to execute that piece of code in both cases.